specification/network/resource-manager/Microsoft.Network/stable/2019-07-01/ipTags.json

```diff
@@ -0,0 +1,406 @@
+        "x-ms-examples": {
+          "Delete_IpTags": {
+            "$ref": "./examples/IpTagsDelete.json"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/ipTags": {
+      "get": {
+        "tags": [
+          "IpTags"
+        ],
+        "operationId": "IpTags_ListByResourceGroup",
+        "description": "Gets all IpTags in a resource group.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of ipTag resources.",
+            "schema": {
+              "$ref": "#/definitions/IpTagsListResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ListByResourceGroup_IpTags": {
+            "$ref": "./examples/IpTagsListByResourceGroup.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Network/ipTags": {
+      "get": {
+        "tags": [
+          "IpTags"
+        ],
+        "operationId": "IpTags_List",
+        "description": "Gets all IpTags in a subscription.",
+        "parameters": [
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of ipTag resources.",
+            "schema": {
+              "$ref": "#/definitions/IpTagsListResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List_IpTags": {
+            "$ref": "./examples/IpTagsListBySubscription.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "IpTagsListResult": {
+      "properties": {
+        "values": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/IpTags"
+          },
+          "description": "The list of iptag information resources."
+        }
+      },
+      "description": "Response for the ListIpTags API service call."
+    },
+    "IpTags": {
+      "properties": {
+        "provisioningState": {
```

Comment:
`provisioningState` should be read-only

Reply:
General Question: How do you determine which properties must be read-only?

Reply:
There are common properties for top-level resources that should be read-only since they contain service information for the customer: `type`, `provisioningState`, `etag`, `resourceGuid`
Also, I check contract models in NRP's backend to see if other props should be read-only

Reply:
Thanks for the comment and doing this code review. I have checked these attributes and made corrrection